### PR TITLE
Refactor/extract branch creation logic

### PR DESCRIPTION
**Pull request by tupe12334**
Close: #5129 

## PR Details

This pull request extract the branch creation logic from the provider to the service
## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Review by abrl91 — CHANGES_REQUESTED** [review]
These changes are related to another PR on which I commented that the change doesn't conform to our conventions.
https://github.com/amplication/amplication/pull/5128
Please fix it first

File: packages/amplication-git-utils/src/git/git.service.ts [hunk 1]
```diff
@@ -124,6 +125,19 @@ export class GitClientService {
     }
   }
 
+  private async createBranchIfNotExists(args: OneBranchArgs) {
```
Comment (abrl91) [inline packages/amplication-git-utils/src/git/git.service.ts:128]: Why did you add this method to the git provider?
@yuval-hazaz when we talked about it while I was working on the refactoring we decided that those functions will be in the provider and not in the git service. What changed

Comment (abrl91) [reply packages/amplication-git-utils/src/git/git.service.ts:128]: https://github.com/amplication/amplication/pull/5128

File: packages/amplication-git-utils/src/types.ts [hunk 2]
```diff
@@ -141,3 +135,19 @@ export interface CreateCommitArgs {
...
 }
+
+export interface OneBranchArgs extends FindOneRepositoryArgs {
+  branchName: string;
+}
+export interface FindOneRepositoryArgs {
+  owner: string;
+  repositoryName: string;
+}
+
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/types.ts:146]: is this being used?

File: packages/amplication-git-utils/src/types.ts [hunk 3]
```diff
@@ -141,3 +135,19 @@ export interface CreateCommitArgs {
   branchName: string;
   files: UpdateFile[];
 }
+
+export interface OneBranchArgs extends FindOneRepositoryArgs {
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/types.ts:139]: Why do we need to extend? What is the relation between OneBranchArgs and FindOneRepositoryArgs? I don't think there is any relation. Args of a function may change in one function without any change in the other, so this relation is not helping and just make things harder

Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/types.ts:139]: please rename OneBranchArgs to getBranchArgs

File: packages/amplication-git-utils/src/types.ts [hunk 4]
```diff
@@ -141,3 +135,19 @@ export interface CreateCommitArgs {
...
+  branchName: string;
+}
+export interface FindOneRepositoryArgs {
+  owner: string;
+  repositoryName: string;
+}
+
+export interface CreateBranchArgs extends OneBranchArgs {
+  pointingSha: string;
+}
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/types.ts:149]: please remove the extends and use all immediate params

File: packages/amplication-git-utils/src/git/git.service.ts (hunk 1 above)
Comment (yuval-hazaz) [reply packages/amplication-git-utils/src/git/git.service.ts:128]: I think this way is ok

File: packages/amplication-git-utils/src/git/git.service.ts [hunk 5]
```diff
@@ -124,6 +125,19 @@ export class GitClientService {
...
+    const isBranchExist = await this.provider.isBranchExists(args);
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/git/git.service.ts:129]: use this.provider.getBranch(args) instead of isBranchExists

File: packages/amplication-git-utils/src/GitProvider.ts [hunk 6]
```diff
@@ -33,14 +35,15 @@ export interface GitProvider {
...
-    createBranchIfNotExistsArgs: CreateBranchIfNotExistsArgs
-  ) => Promise<Branch>;
   createCommit: (createCommitArgs: CreateCommitArgs) => Promise<void>;
   getPullRequestForBranch: (
     getPullRequestForBranchArgs: GetPullRequestForBranchArgs
   ) => Promise<{ url: string; number: number } | undefined>;
   createPullRequestForBranch: (
     createPullRequestForBranchArgs: CreatePullRequestForBranchArgs
   ) => Promise<string>;
+  isBranchExists: (args: OneBranchArgs) => Promise<boolean>;
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/GitProvider.ts:45]: we dont need isBranchExists - just use "getBranch"

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-git-utils/src/GitProvider.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import {
   Branch,
-  CreateBranchIfNotExistsArgs,
+  Commit,
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/GitProvider.ts:3]: please rename the file to `git-provider.interface.ts`

File: packages/amplication-git-utils/src/types.ts (hunk 2 above)
Comment (tupe12334) [reply packages/amplication-git-utils/src/types.ts:146]: Yes

File: packages/amplication-git-utils/src/git/git.service.ts (hunk 5 above)
Comment (tupe12334) [reply packages/amplication-git-utils/src/git/git.service.ts:129]: No because it will throw an error if the branch don't exist

File: packages/amplication-git-utils/src/types.ts (hunk 3 above)
Comment (tupe12334) [reply packages/amplication-git-utils/src/types.ts]: You cant find branch without the repository, so instead of rewrite the owner and the repo we just extends

**Review by yuval-hazaz — APPROVED** [review]
